src/main/kotlin/com/kostas/kafkaStreams/metadata/MetadataService.kt

```diff
@@ -0,0 +1,36 @@
+package com.kostas.kafkaStreams.metadata
+
+import org.apache.kafka.common.serialization.Serializer
+import org.apache.kafka.streams.StreamsConfig
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties
+import org.springframework.kafka.config.StreamsBuilderFactoryBean
+import org.springframework.stereotype.Service
+
+@Service
+class MetadataService(
```

Comment:
This looks similar to a class i've seen in CMS ;)

Reply:
Yeah, I used the same way to handle metadata and query forwarding when the requested key is on a remote host. Was not quite sure how can this be simplified tbh.